packages/eth-providers/src/utils/transactionHelper.ts

```diff
@@ -121,16 +121,28 @@ export const encodeGasLimit = (
   txFee: BigNumber,
   gasPrice: BigNumber,
   gasLimit: BigNumber,
-  usedStorage: BigNumber
+  usedStorage: BigNumber,
+  isTokenTransfer = false,
 ): BigNumber => {
   const rawEthGasLimit = txFee.div(gasPrice);
-  const encodedGasLimit = gasLimit.div(GAS_LIMIT_CHUNK).add(1);
-  const encodedStorageLimit = usedStorage.gt(0) ? Math.ceil(Math.log2(usedStorage.toNumber())) : 0;
+  const encodedStorageLimit = usedStorage.gt(0)
+    ? Math.ceil(Math.log2(usedStorage.toNumber()))
+    : 0;
+
+  let encodedGasLimit = gasLimit.div(GAS_LIMIT_CHUNK).add(1);
+  if (isTokenTransfer) {
+    // for token transfer, need to make sure bbb is even
+    // so when metamask 1.5X gasLimit, it won't affect cc
+    encodedGasLimit = encodedGasLimit.mod(2).eq(0)
+      ? encodedGasLimit
+      : encodedGasLimit.add(1);
+  }
```

Comment:
```
encodedGasLimit = encodedGasLimit.div(1000).add(1).mul(1000)
```
It can also ensure that the last 3 bits are 0, reducing the impact of multiples